Update PR template to enable automatic generation of release notes (#454)
CWorthy-ocean/C-Star: .github/PULL_REQUEST_TEMPLATE.md

# Summary

> <!-- Feel free to remove sections and check-list items irrelevant to this PR -->

Author: ```suggestion
<!-- Feel free to remove sections irrelevant to this PR or leave the default `N/A` content -->
```

## Breaking Changes

> 

Author: ```suggestion
<!-- List any breaking changes to interfaces, changes to inputs or outputs, or procedural changes -->
```

- N/A

## New Features

> 

Author: ```suggestion
<!-- List any new capabilities added -->
```

- N/A

## Security Fixes

> 

Author: ```suggestion
<!-- List any changes resulting in a change of security posture -->
```

- N/A

## Bug Fixes

> 

Author: ```suggestion
<!-- List any behavioral changes resulting from pre-existing code performing in an unexpected manner  -->
```

- N/A

## Improvements

> 

Author: ```suggestion
<!-- List any improvements made to existing code or processes  -->
```

- N/A

## Miscellaneous

> 

Author: ```suggestion
<!-- List any non-code-related changes, such as CI, packaging, or documentation -->
```

- N/A

## Code Review Checklist

> 

Author: ```suggestion
<!-- Feel free to remove check-list items irrelevant to this PR -->
```
